spec/spec.md

```diff
@@ -526,81 +729,245 @@ response when returning `anydata` directly from a resource function.
+```ballerina
+@af:CosmosDBTrigger {connectionStringSetting: "CosmosDBConnection", databaseName: "db1", collectionName: "c1"}
+listener af:CosmosDBListener cosmosEp = new ();
+service "cosmos" on cosmosEp {
+  remote function onUpdated (@http:Payload DBEntry[] inMsg) returns @af:CosmosDBOutput {connectionStringSetting: "CosmosDBConnection", databaseName: "db1", collectionName: "c2"} DBEntry[]|error {
+    return inMsg;
+  }
+}
+```
+#### 3.2.5. TwilioSmsOutput
+[TwilioSmsOutput binding](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-twilio?tabs=in-process%2Cfunctionsv2&pivots=programming-language-java) allows to send messages to the specified telephone number.
+```ballerina
 @af:TimerTrigger { schedule: "*/10 * * * * *" }
 listener af:TimerListener timerEp = new ();
- 
+
 service "timer" on timerEp {
-   remote function onTriggered (@http:Payload json inMsg) returns @af:QueueOutput @af:TwilioSmsOutput { fromNumber: "+12069845840" }  string|error {
-       string id = inMsg[0].id;
-       return "helloo "+ id;
-   }
+  remote function onTriggered (@http:Payload json inMsg) returns @af:QueueOutput @af:TwilioSmsOutput { fromNumber: "+12069845840" }  string|error {
+      string id = inMsg[0].id;
+      return "helloo "+ id;
+  }
+}
+```
+### 3.3. Input bindings
+[Input bindings](https://learn.microsoft.com/en-us/azure/azure-functions/functions-triggers-bindings?tabs=csharp) allows to get inputs from other resources. In Ballerina Azure Function, Input bindings are handled as input parameters with proper annotation and type.
+#### 3.3.1. CosmosDBInput
+[CosmosDBInput binding](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-cosmosdb-v2-input?tabs=in-process%2Cfunctionsv2&pivots=programming-language-java) uses the SQL API to retrieve one or more Azure Cosmos DB documents and passes them to the input parameter of the function.
+```ballerina
+resource function post db(@http:Payload string greeting, @af:CosmosDBInput {
+    connectionStringSetting: "CosmosDBConnection",databaseName: "db1",
+    collectionName: "c2", sqlQuery: "SELECT * FROM Items"} DBEntry[] input1) returns @af:HttpOutput string|error {
+        return "Hello " + greeting + input1[0].id;
+}
+```
+#### 3.3.2. BlobInput
+[BlobInput binding](https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-blob-input?tabs=in-process%2Cextensionv5&pivots=programming-language-java) allows to read blob storage data as input to an Azure Function.
+```ballerina
+resource function get isBlobEmpty(@af:BlobInput { path: "bpath1/newBlob"} byte[]|() input1) returns @af:HttpOutput boolean {
+    if (input1 == ()) {
+        return true;
+    }
+    return false;
+}
+```
+
+## 4. Artifact generation
```

Comment:
Shall we move this section up?
We can make a section mentioning the things the azure functions module do
1. Artifact generation.
2. Runtime routing and data binding.